mscore/musescore.cpp

```diff
@@ -4948,13 +4955,28 @@ int main(int argc, char* av[])
             QString temp = parser.value("T");
             if (temp.isEmpty())
                    parser.showHelp(EXIT_FAILURE);
-            trimMargin = temp.toInt(); // ToDo: check whether conversion went OK, set to -1 if not
+            bool ok = false;
+            trimMargin = temp.toInt(&ok);
+            if (!ok)
+                  trimMargin = -1;
            }
       if (parser.isSet("x")) {
             QString temp = parser.value("x");
             if (temp.isEmpty())
                    parser.showHelp(EXIT_FAILURE);
-            guiScaling = temp.toDouble(); // ToDo: check whether conversion went OK, set to 1.0 if not
+            bool ok = false;
+            guiScaling = temp.toDouble(&ok);
+            if (!ok)
+                  guiScaling = 0.0;
```

Comment:
For the record, 0.0 is the default, and it means the scaling factor will be calculated based on the physical dots per inch reported by Qt, which won't be 1.0 except on 96 DPI displays.  In other words, "-x 0.0" is like saying, "do the right thing" whereas "-x 1.0" is like saying "ignore that you think is right, act as though my display is 96 DPI".  Probably we could reconsider how these calculations are done, so that the argument to -x is *applied to the result* of the automatic calculation (eg, -x 2.0 means twice as large as "normal") rather than *replacing* the automatic calculation.  When I originally wrote the '-x" code, though, we didn't have a concept of automatic calculation of resolution, and I'd just as soon not change the interpretation of the "-x" parameter at this point.